ansible/configs/aap2-ansible-workshops/pre_infra.yml

```diff
@@ -0,0 +1,77 @@
+---
+
+- name: Step 000 Pre Infrastructure
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  become: false
+  vars_files:
+    - "{{ playbook_dir }}/configs/{{ env_type }}/env_vars.yml"
+  tags:
+    - step001
+    - pre_infrastructure
+    - generate_env_keys
+
+  tasks:
+
+    - debug:
+        msg: "Step 000 Pre Infrastructure"
+
+    - name: Generate SSH keys when set_env_authorized_key
+      block:
+
+        - name: Generate SSH keys
+          shell: ssh-keygen -b 2048 -t rsa -f "{{ playbook_dir }}/workdir/{{ env_authorized_key }}" -q -N ""
+          args:
+            creates: "{{ playbook_dir }}/workdir/{{ env_authorized_key }}"
+
+        - name: fix permission
+          file:
+            path: "{{ playbook_dir }}/workdir/{{ env_authorized_key }}"
+            mode: 0400
+
+        - name: Generate SSH pub key
+          shell: >-
+            ssh-keygen -y -f "{{ playbook_dir }}/workdir/{{ env_authorized_key }}"
+            > "{{ playbook_dir }}/workdir/{{ env_authorized_key }}.pub"
+          args:
+            creates: "{{ playbook_dir }}/workdir/{{ env_authorized_key }}.pub"
+      when: set_env_authorized_key | bool
+
+- name: PreSoftware flight-check
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  become: false
+  vars_files:
+    - "{{ playbook_dir }}/configs/{{ env_type }}/env_vars.yml"
+  tags:
+    - flight_check
+  tasks:
+
+    - debug:
+        msg: "Pre-Software checks completed successfully"
+
+- import_playbook: "{{ playbook_dir }}/workdir/{{ env_type }}/provisioner/provision_lab.yml"
+  tags:
+    - ansible_workshops_provision
+
+- name: PostSoftware flight-check
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  become: false
+  vars_files:
+    - "{{ playbook_dir }}/configs/{{ env_type }}/env_vars.yml"
+  tags:
+    - post_flight_check
+
+  tasks:
+
+    - when: create_login_page
+      name: Output user.info string for email when a workshop, create_login_page set to true, S3 bucket
+      debug:
+        msg: "user.info: {{ (student_workloads | default('Ansible')) | regex_replace('_tower_workshop','') | regex_replace('_', ' ') | title }} Workshop and Student VMs are available at: http://{{ guid }}.{{ workshop_dns_zone }}"
```

Comment:
Need to use `agnosticd_user_info`.

Reply:
I'd like to defer that we are throwing this away soon. This code clearly works as it is a like for like copy of the `ansible-workshops` code? This will never go to Babylon, as they are creating a new Babylon/agd native config

Reply:
Cleaned up to use agnosticd_user_info